client/src/pages/Explorer.js

```diff
@@ -1,68 +1,93 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import Typography from "@material-ui/core/Typography";
 import { makeStyles } from "@material-ui/core/styles";
-//Remove the next line when backend data is available.
-import places from "../database/places";
 import Grid from "@material-ui/core/Grid";
 import Container from "@material-ui/core/Container";
 import Favorite from "@material-ui/icons/Favorite";
-import {CustomSmallerCheckBox} from "../themes/theme";
+import { CustomSmallerCheckBox } from "../themes/theme";
 
 const useStyles = makeStyles({
-	pageContainer: {
-		display: "flex",
-		flexDirection: "column",
-		alignItems: "center",
-		marginTop: 20,
-		padding: "0 20px",
-	},
-	paperContainer: {
-		height: 300,
-		borderRadius: 10,
-		backgroundRepeat: "no-repeat",
-		backgroundSize: "cover",
-		display: "flex",
-		flexDirection: "column",
-		justifyContent: "flex-end",
-		overflow: "hidden",
-	},
-	title: {
-		margin: 5,
-		textAlign: "center",
-	},
-	customCheckBoxRoot: {
-		width: "5px",
-		height: "5px",
-	},
-	bottomInformationContainer: {
-		display: "flex",
-		height: "15%",
-		borderTop: "1px solid #a9a9a9",
-		padding: 10,
-		justifyContent: "space-between",
-	},
-	bottomInformationSubContainer1: {
-		display: "flex",
-		flexDirection: "column",
-		justifyContent: "center",
-		paddingLeft: 10,
-	},
-	bottomInformationSubContainer2: {
-		display: "flex",
-		flexDirection: "column",
-		justifyContent: "center",
-		alignItems: "flex-end",
-	},
+  pageContainer: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    marginTop: 20,
+  },
+  paperContainer: {
+    height: 250,
+    width: 195,
+    borderRadius: 10,
+    backgroundRepeat: "no-repeat",
+    backgroundSize: "cover",
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "flex-end",
+    overflow: "hidden",
+  },
+  title: {
+    margin: 5,
+    textAlign: "center",
+  },
+  customCheckBoxRoot: {
+    width: "5px",
+    height: "5px",
+  },
+  bottomInformationContainer: {
+    display: "flex",
+    height: "15%",
+    borderTop: "1px solid #a9a9a9",
+    padding: 10,
+    justifyContent: "space-between",
+  },
+  bottomInformationSubContainer1: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    paddingLeft: 10,
+  },
+  bottomInformationSubContainer2: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "flex-end",
+  },
 });
+let userFavoritePlaces = [];
+function handleChange(checked, selectedElement) {
+  if (checked) {
+    //add
+    userFavoritePlaces.push(selectedElement.name);
+  } else {
+    //remove
+    const placeIndex = userFavoritePlaces.indexOf(selectedElement.name);
+    if (placeIndex >= 0) {
+      userFavoritePlaces.splice(placeIndex, 1);
+    }
+  }
+  fetch("/api/users/updateFavoriteCities", {
```

Comment:
this route is probably too specifically named, also try to avoid camelCase routes. Usually when making REST routes, we do something like `/api/users/:userId/favorite-cities`. Now on the backend we can define what a `POST` request to this route means and in our case it's to update them.

Reply:
Ok, noted. Thanks

Question:
Assuming that only authenticated user can update favorites and favorites can only be updated by the owner user, is it necessary to pass the :userId as required parameter here as the user information can be fetched with the JWT token?